fsm.go

```diff
@@ -127,17 +129,44 @@ func (r *fsmS) announceSensor(e *f.Event) {
 	log.debug("announcing sensor to the agent")
 
 	go func(cb func(b bool, from *FromS)) {
+
 		d := &Discovery{
 			PID:  os.Getpid(),
 			Name: os.Args[0],
 			Args: os.Args[1:]}
 
+		var socket net.Conn
```

Comment:
How do you make sure that this socket is used for the HTTP request? Looks like magic here :)